controller/config.go

```diff
@@ -857,11 +857,12 @@ func validateCAASImageRepo(imageRepo string) (string, error) {
 		return "", errors.Trace(err)
 	}
 	if imageDetails.IsPrivate() {
-		r, err := registry.NewRegistry(*imageDetails)
+		r, err := registry.New(*imageDetails)
 		if err != nil {
 			return "", errors.Trace(err)
 		}
 		*imageDetails = r.ImageRepoDetails()
+		func() { _ = r.Close() }()
```

Comment:
why the func?

Reply:
Ah, I think I missed defer, but it doesn't really matter.